git/state.go

```diff
@@ -0,0 +1,63 @@
+package git
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
```

Comment:
I am aware of this.  did you see the comment I left at the top?

> // READONLY-NOTE: this may belong in its own package ...

I left it as is knowingly to avoid further gutting the git functionality before we agree on the general design.

Reply:
resolving since the native state syncing has by now moved out of this package